### PR TITLE
✨  Add notes access to zoho modules

### PR DESCRIPTION
**Pull request by romain-verbeke**
https://www.zoho.com/crm/developer/docs/api/v2/get-notes.html

This PR need to be merged so i can update the gemfile here later: https://github.com/wecasa/web/pull/3663/files

### CODE REVIEW

File: lib/zoho_hub/base_record.rb [hunk 1]
```diff
@@ -241,4 +250,13 @@ def build_response(body)
       self.class.build_response(body)
     end
   end
+
+  class Note < BaseRecord
```
Comment (Fabien-Wecasa) [inline lib/zoho_hub/base_record.rb]: An idea to lighten this huge class: create a module, like we did [for the attachments](https://github.com/Fabien-Wecasa/zoho_hub/pull/5/files), put in the `Note` class and related method (`notes` and `add_note`),  and `require` it in `zoho_hub.rb`.

File: lib/zoho_hub/base_record.rb [hunk 2]
```diff
@@ -241,4 +250,6 @@ def build_response(body)
       self.class.build_response(body)
     end
   end
+
+  require 'zoho_hub/modules/note'
```
Comment (romain-verbeke) [inline lib/zoho_hub/base_record.rb]: @Fabien-Wecasa I moved the code into its own file but i do need to require the module after the ZohoHub::BaseRecord is defined

Comment (Fabien-Wecasa) [reply lib/zoho_hub/base_record.rb]: Sorry, misunderstanding: the related functions `notes` and `add_note` should also be moved in the module, and I guess you won't need this require anymore. (see the link in my previous comment)

Comment (Fabien-Wecasa) [reply lib/zoho_hub/base_record.rb]: https://github.com/Fabien-Wecasa/zoho_hub/pull/5/files#diff-c28dcd616422d5d722bb0cb44e786ba5dc1064e1d52cc2ab489e22cd3ec98f13

**Pull request merged**